src/ripple/legacy/0.27/book/Amount.h

```
@@ -0,0 +1,216 @@
+        : m_mantissa (mantissa)
+        , m_exponent (exponent)
+        , m_negative (negative)
+        , m_integral (integral)
+    {
+    }
+
+public:
+    /** Default construction.
+        The value is uninitialized.
+    */
+    AmountType() noexcept
+    {
+    }
+
+    /** Construct from an integer.
+        The representation is set to integral.
+    */
+    /** @{ */
+    template <class Integer>
+    AmountType (Integer value,
+        std::enable_if_t <std::is_signed <Integer>::value>* = 0) noexcept
+        : m_mantissa (value)
+        , m_exponent (0)
+        , m_negative (value < 0)
+        , m_integral (true)
+    {
+        static_assert (std::is_integral<Integer>::value,
+            "Cannot construct from non-integral type.");
+    }
+
+    template <class Integer>
+    AmountType (Integer value,
+        std::enable_if_t <! std::is_signed <Integer>::value>* = 0) noexcept
+        : m_mantissa (value)
+        , m_exponent (0)
+        , m_negative (false)
+    {
+        static_assert (std::is_integral<Integer>::value,
+            "Cannot construct from non-integral type.");
+    }
+    /** @} */
+
+    /** Assign the value zero.
+        The representation is preserved.
+    */
+    AmountType&
+    operator= (Zero) noexcept
+    {
+        m_mantissa = 0;
+        // VFALCO Why -100?
+        //        "We have to use something in range."
+        //        "This makes zero the smallest value."
+        m_exponent = m_integral ? 0 : -100;
+            m_exponent = 0;
+        m_negative = false;
+        return *this;
+    }
+
+    /** Returns the value in canonical format. */
+    AmountType
+    normal() const noexcept
+    {
+        if (m_integral)
+        {
+            AmountType result;
+            if (m_mantissa == 0)
+            {
+                result.m_exponent = 0;
+                result.m_negative = false;
+            }
+            return result;
+        }
+        return AmountType();
```

That seems dodgy - you silently return an uninitialized value?